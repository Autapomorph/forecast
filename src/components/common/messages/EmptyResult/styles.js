import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledEmptyListMessage = styled(Message)``;
