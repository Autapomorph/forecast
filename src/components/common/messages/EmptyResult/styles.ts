import styled from 'styled-components/macro';

import Message from 'components/common/messages/Base';

export const EmptyResultMessage = styled(Message)`
  margin: 20px 0;
`;
