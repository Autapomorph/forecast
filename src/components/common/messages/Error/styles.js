import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
  font-size: 1.3rem;
`;
