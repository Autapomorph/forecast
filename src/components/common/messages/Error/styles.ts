import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
`;
