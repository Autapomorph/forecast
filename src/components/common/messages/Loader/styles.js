import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledLoaderMessage = styled(Message)`
  margin: 10px 0;
`;
