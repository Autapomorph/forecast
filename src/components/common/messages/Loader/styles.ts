import styled from 'styled-components/macro';

import Message from 'components/common/messages/Base';

export const LoaderMessage = styled(Message)`
  margin: 10px 0;
  font-size: 1.5rem;
`;
