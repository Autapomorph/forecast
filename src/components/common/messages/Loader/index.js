import React from 'react';

import { StyledLoaderMessage } from './styles';

const Loader = ({ children, ...props }) => (
  <StyledLoaderMessage {...props}>{children}</StyledLoaderMessage>
);

Loader.defaultProps = {
  children: 'Загрузка',
};

export default Loader;
