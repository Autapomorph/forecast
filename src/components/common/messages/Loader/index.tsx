import React from 'react';
import MDSpinner from 'react-md-spinner';

import { StyledLoaderMessage } from './styles';

const Loader: React.FC = ({ children, ...props }): React.ReactElement => (
  <StyledLoaderMessage {...props}>{children}</StyledLoaderMessage>
);

Loader.defaultProps = {
  children: <MDSpinner singleColor="#fff" />,
};

export default Loader;
