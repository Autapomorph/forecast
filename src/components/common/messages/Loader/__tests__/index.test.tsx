import React from 'react';
import { shallow } from 'enzyme';

import Loader from '..';

import * as S from '../styles';

describe('Loader tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Loader>Test</Loader>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const text = 'text';
    const wrapper = shallow(<Loader>{text}</Loader>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.LoaderMessage).text()).toEqual(text);
  });

  it('should render default children', () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(Loader.defaultProps.children));
  });
});
