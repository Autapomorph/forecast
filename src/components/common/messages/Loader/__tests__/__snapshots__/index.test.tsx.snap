// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader tests renders properly 1`] = `
<styles__LoaderMessage>
  Test
</styles__LoaderMessage>
`;

exports[`Loader tests should render children 1`] = `
<styles__LoaderMessage>
  text
</styles__LoaderMessage>
`;

exports[`Loader tests should render default children 1`] = `
<styles__LoaderMessage>
  <MDSpinner
    color1="rgb(66, 165, 245)"
    color2="rgb(239, 83, 80)"
    color3="rgb(253, 216, 53)"
    color4="rgb(76, 175, 80)"
    duration={1333}
    singleColor="#fff"
    size={28}
  />
</styles__LoaderMessage>
`;
